### ISSUE TITLE
Post request is being cached when submitting a form and super cache is enabled via .htaccess

### ISSUE THREAD

**Issue by theodejager** [Automattic/jetpack#35112]
### Impacted plugin

Super Cache

### Quick summary

When submitting a form (via Formidable Forms) on a frontend page, sometimes the post request containing the form data returns the cached page and the form isn't submitted. Not 100% sure if this is a bug here or on their end (didn't test a plain custom form yet), but looking at the network tab the submit does a POST request to the page, and the returned page has the cache info in the source:

```
<!-- Dynamic page generated in 0.194 seconds. -->
<!-- Cached page generated by WP-Super-Cache on 2024-01-18 14:45:27 -->

<!-- super cache -->
```

This is with Super Cache enabled via .htaccess. Adding a rule excluding the form page in the settings, like "/contact/" fixes the issue. 

### Steps to reproduce

1. I have a page with a Formidable shortcode in it.
2. After submitting the form, a post request to the same page is seen in the network tab
3. The reloaded page is a cached version of the form, that was super cached previously, and the form misses the submission.

### A clear and concise description of what you expected to happen.

Post request should not be cached?

### What actually happened

Post request is cached

### Impact

Some (< 50%)

### Available workarounds?

Yes, easy to implement

**Comment by donnchawp** [comment]
Did you have "late init" enabled in the settings when this happened? I fixed a bug in #36211 that would show a cached page after a POST request, but only if "late init" was enabled. This may be what caused the problem you saw in January.

Have you updated to the latest WP Super Cache? Does the problem still occur?

**Comment by donnchawp** [comment]
Closing as I'm pretty sure this was fixed due to another report.
